dpd/p4/sidecar.p4

```diff
@@ -373,66 +490,68 @@ control NatIngress (
 		actions = { forward_ipv6_to; }
 
 		const size = IPV6_NAT_TABLE_SIZE;
-		counters = ipv6_ingress_counter;
+		counters = ipv6_ingress_ctr;
 	}
 
-	// The following actions and table are used to generate the final
-	// "length" field in the ipv4 pseudo header, which needs to be backed
-	// out of the inner udp/tcp checksums to find the residual for the
-	// packet body.  This seems ludicrously complicated, but it's the only
-	// way I've found to do the calculation without running afoul of
-	// limitations in p4 and/or tofino, governing exactly how much work
-	// can be done in each stage and which PHV fields you are allowed
-	// to access.  We are using the 'add' action to subtract the size of
-	// the IPv4 header.  Why?  Because the p4 compiler will let me add a
-	// parameter in an action, but will only let me subtract a constant.
-	// So, I can create a single action that will add the negative
-	// parameter I've manually computed, or I can create 11 actions, each
-	// of which will subtract a hard-coded constant.  Either seems stupid,
-	// but here we are.
-	// XXX: find a less stupid solution
-	action invert() {
-		meta.l4_length = ~meta.l4_length;
+	/* Invert the source and destination addresses.
+
+	*
+	 * This is a separate file for reusability.
```

Comment:
Can this be restructured to have a control that is instantiated in multiple places instead of using the preprocessor to kersplat a few actions and a table definition in place?